gooddata-pandas/gooddata_pandas/result_convertor.py

```diff
@@ -0,0 +1,315 @@
+
+    :param response: execution response to work with;
+    :return:
+    """
+    num_dims = len(response.dimensions)
+    offset = [0 for _ in range(num_dims)]
+    limit = [page_size for _ in range(num_dims)]
+    acc = _AccumulatedData()
+
+    while True:
+        # top-level loop pages through the first dimension;
+        #
+        # if one-dimensional result, it pages over an array of data
+        # if two-dimensional result, it pages over table rows
+        result = response.read_result(offset=offset, limit=limit)
+
+        acc.accumulate_data(from_result=result)
+        acc.accumulate_headers(from_result=result, from_dim=0)
+        acc.accumulate_grand_totals(from_result=result, paging_dim=0, response=response)
+
+        if num_dims > 1:
+            # when result is two-dimensional make sure to read the column headers and column totals
+            # just once - when scrolling 'to the right' for the first time;
+            load_headers_and_totals = False
+            if acc.data_headers[1] is None:
+                acc.accumulate_headers(from_result=result, from_dim=1)
+                load_headers_and_totals = True
+
+            if not result.is_complete(dim=1):
+                # have two-dimensional result (typical table) and the page does not contain
+                # all the columns.
+                #
+                # page 'to the right' to get data from all columns, extend existing rows with that data
+                offset = [offset[0], result.next_page_start(dim=1)]
+                while True:
+                    result = response.read_result(offset=offset, limit=limit)
+                    acc.extend_existing_row_data(from_result=result)
+
+                    if load_headers_and_totals:
+                        acc.accumulate_headers(from_result=result, from_dim=1)
+                        acc.accumulate_grand_totals(from_result=result, paging_dim=1, response=response)
+
+                    if result.is_complete(dim=1):
+                        break
+
+                    offset = [offset[0], result.next_page_start(dim=1)]
+
+        if result.is_complete(dim=0):
+            break
+
+        offset = [result.next_page_start(dim=0), 0] if num_dims > 1 else [result.next_page_start(dim=0)]
+
+    return acc.result()
+
+
+def _create_header_mapper(
+    response: BareExecutionResponse, dim: int, label_overrides: LabelOverrides = {}
+) -> Callable[[int, Any], str]:
+    """
+    Prepares header mapper function which is doing header structures translations into appropriate label used
+    in a dataframe
+    :param response: Response structure to gather dimension header details
+    :param dim: dimension id
+    :param label_overrides: label overrides
+    :return: Mapper function
+    """
+    dim_descriptor = response.dimensions[dim]
+    attribute_labels = label_overrides["labels"] if "labels" in label_overrides else {}
+    measure_labels = label_overrides["metrics"] if "metrics" in label_overrides else {}
```

Comment:
Please use 
attribute_labels = label_overrides.get("labels", {})

Reply:
fixed in last commit